src/amd/compiler/aco_lower_bool_phis.cpp

```diff
@@ -206,34 +207,27 @@ void lower_linear_bool_phi(Program *program, Block *block, aco_ptr<Instruction>&
 void lower_bool_phis(Program* program)
 {
    for (Block& block : program->blocks) {
-      std::vector<aco_ptr<Instruction>> instructions;
-      std::vector<aco_ptr<Instruction>> non_phi;
-      instructions.swap(block.instructions);
-      block.instructions.reserve(instructions.size());
-      unsigned i = 0;
-      for (; i < instructions.size(); i++)
-      {
-         aco_ptr<Instruction>& phi = instructions[i];
-         if (phi->opcode != aco_opcode::p_phi && phi->opcode != aco_opcode::p_linear_phi)
-            break;
-         if (phi->opcode == aco_opcode::p_phi && phi->definitions[0].regClass() == s2) {
-            non_phi.emplace_back(std::move(lower_divergent_bool_phi(program, &block, phi)));
-         } else if (phi->opcode == aco_opcode::p_linear_phi && phi->definitions[0].regClass() == s1) {
+      /* find the first non phi instruction */
+      auto IsPhi = [] (aco_ptr<Instruction>& phi) -> bool { return is_phi(phi); };
+      auto end = std::find_if_not(block.instructions.begin(), block.instructions.end(), IsPhi);
+
+      /* partition divergent boolean phi's to the back */
+      auto IsNotDivergentBooleanPhi = [] (aco_ptr<Instruction>& phi) -> bool {
+         return !(phi->opcode == aco_opcode::p_phi && phi->definitions[0].regClass() == s2);
+      };
+      std::stable_partition(block.instructions.begin(), end, IsNotDivergentBooleanPhi);
+
+      /* lower the boolean phi's */
+      const size_t endSize = std::distance(block.instructions.begin(), end);
```

Comment:
nitpick: we use snake_case for variables, so this should be `end_size`

Reply:
Yeah habbits are hard to break

Reply:
Fixed locally